scripts/build-renode.sh

```diff
@@ -0,0 +1,265 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+source $SCRIPT_DIR/settings.sh
+
+set -x
+set -e
+
+RENODE_BIN="renode"
+command -v $RENODE_BIN 2>&1 1>/dev/null
+if [ $? -ne 0 ]; then
```

Comment:
Please add a comment here. I'm assuming this is downloading a prebuilt renode release? On what platforms will this run?

Reply:
You are right, this is a pre-built Renode compiled for 64-bit Linux. It should run on any distro assuming dependencies are present in the system (list of runtime dependecies is described [here](https://github.com/renode/renode/blob/master/README.rst#installation).

Other option is to install package (that would take care of dependencies automatically). The downside is that:

- it's installed globally in the system, polluting it a little bit,
- packages are distro-specific and we currently support: debian/ubuntu, fedora and arch so we would have to detect the distribution and install appropriate package; the question what about other distributions?